src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -214,35 +209,36 @@ object Gen extends GenArities with GenVersionSpecific {
 
   private[scalacheck] trait R[+T] {
     def labels: Set[String] = Set()
-    def sieve[U >: T]: U => Boolean = _ => true
+    // sieve is no longer used but preserved for binary compatibility
+    final def sieve[U >: T]: U => Boolean = (_: U) => true
     protected def result: Option[T]
     def seed: Seed
 
-    def retrieve: Option[T] = result.filter(sieve)
+    def retrieve: Option[T] = result
 
     def copy[U >: T](
       l: Set[String] = this.labels,
+      // s is no longer used but preserved for binary compatibility
       s: U => Boolean = this.sieve,
       r: Option[U] = this.result,
       sd: Seed = this.seed
     ): R[U] = new R[U] {
       override val labels = l
-      override def sieve[V >: U] = { (x: Any) =>
-        try s(x.asInstanceOf[U])
-        catch { case _: java.lang.ClassCastException => false }
-      }
       val seed = sd
       val result = r
     }
 
-    def map[U](f: T => U): R[U] = r(retrieve.map(f), seed).copy(l = labels)
+    def map[U](f: T => U): R[U] =
+      r(retrieve.map(f), seed).copy(l = labels)
 
-    def flatMap[U](f: T => R[U]): R[U] = retrieve match {
-      case None => r(None, seed).copy(l = labels)
-      case Some(t) =>
-        val r = f(t)
-        r.copy(l = labels ++ r.labels, sd = r.seed)
-    }
+    def flatMap[U](f: T => R[U]): R[U] =
+      retrieve match {
+        case None =>
+          r(None, seed).copy(l = labels)
+        case Some(t) =>
+          val r = f(t)
+          r.copy(l = labels ++ r.labels, sd = r.seed)
```

Comment:
would `labels | r.labels` be clearer that this is set union (and possibly faster)?

Reply:
Yes and possibly -- seems like a good idea.